File: 383-ransom-note/383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        map = {}
        
        for c in magazine:
            map[c] = map.get(c, 0) + 1
        
        for c in ransomNote:
            val = map.get(c, None)
            if not val or val == 0:
                return False
            map[c] -= 1
        
        return True

File: 383-ransom-note/383-ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int[] arr = new int[255];
        
        // store magazine inside map
        for (int i = 0; i < magazine.length(); i++) {
            char c = magazine.charAt(i);
            arr[c] = arr[c] + 1;
        }
        
        for (int i = 0; i < ransomNote.length(); i++) {
            char c = ransomNote.charAt(i);
            
            if (arr[c] > 0) {
                arr[c] = arr[c] - 1;
            } else {
                return false;
            }
        }
        
        return true;
    }
}